classes/Customer.php

```diff
@@ -1193,24 +1193,22 @@ public function transformToCustomer($idLang, $password = null)
 
         $this->is_guest = false;
 
+        /** @var \PrestaShop\PrestaShop\Core\Crypto\Hashing $crypto */
+        $crypto = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Core\\Crypto\\Hashing');
+
         /*
         * If this is an anonymous conversion and we want the customer to set his own password,
-        * we set a random one for now.
-        * TODO - This should be revised in the future because 16 chars can be outside of bounds of
-        * isAcceptablePasswordLength. It should not be checked.
+        * we set a random one for now. If a password was provided, we check it's validity.
         */
         if (empty($password)) {
-            $password = Tools::passwdGen(16, 'RANDOM');
-        }
-
-        if (!Validate::isAcceptablePasswordLength($password) || !Validate::isAcceptablePasswordScore($password)) {
-            return false;
+            $this->passwd = $crypto->hash(Tools::passwdGen(16, 'RANDOM'));
```

Comment:
```suggestion
            $passwordGenerator = new \PrestaShop\PrestaShop\Core\Security\PasswordGenerator();
            $this->passwd = $crypto->hash($passwordGenerator->generatePassword(16, 'RANDOM'));
```

Reply:
This file doesnt exist, passwdGen is not deprecated and used widely in the core

Reply:
It is, just rebase your code :D

Reply:
Better use migrated function instead of legacy ones :)

Reply:
Yes, but how is 
`new PasswordGenerator(new OpenSSL()))->generatePassword(16, 'RANDOM')`
better than
`Tools::passwdGen(16, 'RANDOM')`
?

Reply:
`Tools::passwdGen()` is  legacy. It needs its own autoloader, it is poorly tested and has a poor API.

The new password generator is migrated, 100% unit tested. Has strict types, exception support, etc. For now I agree that this is a little bit longer, but with the upcoming symfony front end container, it will be shorter. We are still migrating step by step, so adding legacy usages is basically slowing down the process.

By adding it into this function you're easing the migration process when we'll migrate the legacy method you are editing. That's one less thing to think about.

Reply:
We can do it in another PR, I am just moving that line in this PR. 👍